eZ/Publish/Core/MVC/Symfony/Controller/Content/ViewController.php

```diff
@@ -102,41 +99,9 @@ protected function buildResponse($etag = null, DateTime $lastModified = null)
      *
      * @deprecated Since 6.0.0. Viewing locations is now done with ViewContent.
      */
-    public function viewLocation($locationId, $viewType, $layout = false, array $params = array())
+    public function viewLocation(View $view)
```

Comment:
I know typehinting would work without custom controllers, but what about custom ones, which call renderLocation/renderContent directly at the end of method? You wouldn't have access to View object in the ViewController then.